include/cuda/std/detail/__atomic

```diff
@@ -190,46 +200,42 @@ struct __cxx_atomic_base_impl_default {
   constexpr __cxx_atomic_base_impl_default() noexcept = default;
   __host__ __device__ constexpr explicit __cxx_atomic_base_impl_default(_Tp __value) noexcept : __a_value(__value) {
   }
-#ifdef __CUDA_ARCH__
   __cxx_atomic_alignment_wrapper_t<_Tp> __a_value;
-#else
-  ::std::atomic<__cxx_atomic_alignment_wrapper_t<_Tp>> __a_value;
-#endif
 };
 
 template<class _Tp, int _Sco>
 __host__ __device__ inline void __cxx_atomic_init(__cxx_atomic_base_impl_default<_Tp, _Sco> volatile* __a, _Tp __val) {
-#if   defined(__CUDA_ARCH__)
     auto __tmp = __cxx_atomic_alignment_wrap(__val);
     __cxx_atomic_assign_volatile(__a->__a_value, __tmp);
-#elif defined(_LIBCUDACXX_CUDA_HAS_NO_HOST_STD_ATOMIC_INIT)
-    __a->__a_value.store(__val, ::std::memory_order_relaxed);
-#else
-    ::std::atomic_init(&__a->__a_value, __cxx_atomic_alignment_wrap(__val));
-#endif
 }
 template<class _Tp, int _Sco>
 __host__ __device__ inline void __cxx_atomic_store(__cxx_atomic_base_impl_default<_Tp, _Sco> volatile* __a, _Tp __val, int __order) {
 #ifdef __CUDA_ARCH__
     detail::__atomic_store_n_cuda(&__a->__a_value, __cxx_atomic_alignment_wrap(__val), __order, detail::__scope_tag<_Sco>());
 #else
-    ::std::atomic_store_explicit(&__a->__a_value, __cxx_atomic_alignment_wrap(__val), (::std::memory_order)__order);
+    __atomic_store(&__a->__a_value, &__val, __order);
 #endif
 }
 template<class _Tp, int _Sco>
 __host__ __device__ inline _Tp __cxx_atomic_load(__cxx_atomic_base_impl_default<_Tp, _Sco> const volatile* __a, int __order) {
 #ifdef __CUDA_ARCH__
     return __cxx_atomic_alignment_unwrap(detail::__atomic_load_n_cuda(&__a->__a_value, __order, detail::__scope_tag<_Sco>()));
 #else
-    return __cxx_atomic_alignment_unwrap(::std::atomic_load_explicit(&__a->__a_value, (::std::memory_order)__order));
+    alignas(_Tp) unsigned char __buf[sizeof(_Tp)];
```

Comment:
@griwes it looks like I tore off another alignment unwrap here.

Are these `char*` output buffers a code smell? This was done to fix initialization warnings.

Reply:
Why not use __atomic_load_n here instead?

Reply:
__atomic_load_n does not work for non-integral non-pointer types: https://gcc.godbolt.org/z/osrcs1

Maybe I'm mistaken in some way about its usage?